File: python/main/edrm/DirectoryEntry.py
import hashlib
from pathlib import Path

import edrm.EDRMUtilities as eutes
from edrm import FieldFactory
from edrm.EntryField import EntryField
from edrm.FileEntry import FileEntry


class DirectoryEntry(FileEntry):
    def __init__(self, directory_path: str, parent_id: str = None):
        self.__directory = Path(directory_path)
        super().__init__(directory_path, 'filesystem/directory', parent_id)

    def fill_hash_fields(self) -> None:
        self['SHA-1'] = FieldFactory.generate_field('SHA-1',
                                                    EntryField.TYPE_TEXT,
                                                    eutes.hash_directory(self.directory, hashlib.sha1()))

    @property
    def directory(self) -> Path:
        return self.__directory

    def add_as_parent_path(self, existing_path: str):
        return self.name + '/' + existing_path

    def calculate_md5(self) -> str:
        return eutes.hash_directory(self.file_path, hashlib.md5())

File: java/src/main/java/com/nuix/nli/edrm/DirectoryEntry.java
package com.nuix.nli.edrm;

import java.io.UnsupportedEncodingException;
import java.net.URLEncoder;
import java.nio.charset.StandardCharsets;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.Map;

import org.w3c.dom.Document;
import org.w3c.dom.Element;

public class DirectoryEntry extends FileEntry {
    private final Path directory;

    public DirectoryEntry(String directoryPath) { this(directoryPath, null); }

    public DirectoryEntry(String directoryPath, String parentId) {
        super(directoryPath, "filesystem/directory", parentId);
        this.directory = Paths.get(directoryPath);
    }

    @Override protected void fillHashFields() {
        setFieldValue("SHA-1", EDRMUtilities.hashDirectory(this.directory, "SHA-1").hashString());
    }

    public Path getDirectory() { return directory; }

    @Override public String addAsParentPath(String existingPath) {
        try {
            return URLEncoder.encode(this.getName(), EDRMUtilities.EDRM_CONFIG
                    .getOrDefault("encoding", StandardCharsets.UTF_8.name())) + "/" + existingPath;
        } catch (UnsupportedEncodingException e) {
            throw new RuntimeException(e);
        }
    }

    @Override public void addFile(Document document, Element container, Map<String, EntryInterface> entryMap, boolean forNli) {
        return;
    }

    @Override protected String calculateMD5() {
        return EDRMUtilities.hashDirectory(this.directory, "MD5").hashString();
    }
}